lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -90,18 +101,35 @@ public function query()
 
         try {
             if ($argsCount == 4) {
-                return parent::query($args[0], $args[1], $args[2], $args[3]);
+                $stmt = parent::query($args[0], $args[1], $args[2], $args[3]);
+
+                $this->trackLastInsertId();
+
+                return $stmt;
             }
 
             if ($argsCount == 3) {
-                return parent::query($args[0], $args[1], $args[2]);
+                $stmt = parent::query($args[0], $args[1], $args[2]);
+
+                $this->trackLastInsertId();
+
+                return $stmt;
             }
 
             if ($argsCount == 2) {
-                return parent::query($args[0], $args[1]);
+                $stmt = parent::query($args[0], $args[1]);
+
+                $this->trackLastInsertId();
```

Comment:
Do these lines need to be duplicated? If the case with `$args[0]` is wrapped in the `else` block, you can use a single return statement and call to `$this->trackLastInsertId();` prior to that.

Reply:
I think our CS doesn't even allow that anymore, but I'll check